home/utils.py

```diff
@@ -67,6 +70,26 @@
 ]
 
 
+def cosine_similarity(A, B):
+    return float((np.dot(A, B) / (np.linalg.norm(A) * np.linalg.norm(B))))
+
+
+def retrieve_top_n_content_pieces(user_input, queryset, n=5, content_type=None):
+    # similar_embeddings = [{'faq_name':, 'faq_content':, 'embedding':}, ...] # We need to filter by content type and then retrieve their embeddings
+    # Generate embedding for user text
+    user_embedding = model.encode([user_input])
+
+    documents_retrieved = []
+    for page in queryset:
```

Comment:
In this for loop, I want to introduce a confidence threshold.

If the highest similarity_score found is less than 0.1, we need rather return a message to the user that explains that we do not have an answer for their question, and perhaps they can try to ask it differently. If the highest similarity_score found is more than 0.1 but less than 0.2, we can return the top n content pieces, but we should also add a disclaimer that we aren't very confident in our response, and again, that the user may want to try and ask the question differently.

Reply:
We decided not to implement the confidence threshold at this time